spring-cloud-deployer-kubernetes/src/main/java/org/springframework/cloud/deployer/spi/kubernetes/DeploymentPropertiesResolver.java

```diff
@@ -173,6 +173,21 @@ Map<String, Quantity> deduceResourceLimits(Map<String, String> kubernetesDeploye
 			cpu = properties.getLimits().getCpu();
 		}
 
+		String ephemeralStorage = PropertyParserUtils.getDeploymentPropertyValue(kubernetesDeployerProperties, this.propertyPrefix + ".limits.ephemeral-storage");
```

Comment:
I know the precedent was already here in this file, but I think we can reduce quite a bit of this deduction boiler plate w/ a helper function that would ease this to something like:
```java
String ephemeralStorage = deducePropertyValue(deployerProps, ".limits.ephemeral-storage",
		() -> this.properties.getLimits().getEphemeralStorage());

String hugePages2Mi = deducePropertyValue(deployerProps, ".limits.hugepages-2Mi",
		() -> this.properties.getLimits().getHugepages2Mi());

String hugePages1Gi = deducePropertyValue(deployerProps, ".limits.hugepages-1Gi",
		() -> this.properties.getLimits().getHugepages1Gi());
```
w/ function looking like:
```java
private <T> String deducePropertyValue(Map<String, String> deployerProps, String propNameSansPrefix, Supplier<String> defaultValue) {
	String propName = this.propertyPrefix + propNameSansPrefix;
	String propValue = PropertyParserUtils.getDeploymentPropertyValue(deployerProps, propName);
	if (!StringUtils.hasText(propValue)) {
		propValue = defaultValue.get();
	}
	return propValue;
}
```
Wdyt?

Reply:
Good suggestion, added